### PR TITLE
Added blog, plus a blog post

### PR DESCRIPTION
**Pull request by dotsdl**
Added a blog post about upcoming plans to release. I plan to do a straight-up release of 0.8.0. Please let me know if you think any more needs to be said in this post.

### CODE REVIEW

**Comment by richardjgowers** [comment]
@dotsdl looks ok, it'll need a section on how to migrate (once we figure this out ourselves :D)

File: _posts/2018-05-06-datreant-downsize.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+## Apologies in advance
+
+Open-source software is hard to produce, and it can be even harder to maintain.
+In an ideal world, we could roll out gradual changes to `datreant` in a controlled way, according to a clear project plan.
+None of our changes would be a huge disruption to users, and we'd do everything technically possible to ease the transition.
+But this project is small, with ~3 developers spending a few extra minutes on it every month.
+We don't have the bandwidth to do everything the way we want to.
+
+With that, we're releasing `datreant` 0.8.0 in the next few days, with a package on PyPI shortly after.
```
Comment (kain88-de) [inline _posts/2018-05-06-datreant-downsize.md]: Please call it datreant 1.0.0

That fits better what we have and avoids headaches with the conda-forge 1.0 stub package.

File: _posts/2018-05-06-datreant-downsize.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+This functionality, however, was difficult to maintain, and writing new limbs wasn't actually easy to do.
+What's more, building a CLI interface for Treants was confounded by this complexity.
+
+## Apologies in advance
+
+Open-source software is hard to produce, and it can be even harder to maintain.
+In an ideal world, we could roll out gradual changes to `datreant` in a controlled way, according to a clear project plan.
+None of our changes would be a huge disruption to users, and we'd do everything technically possible to ease the transition.
+But this project is small, with ~3 developers spending a few extra minutes on it every month.
+We don't have the bandwidth to do everything the way we want to.
```
Comment (kain88-de) [inline _posts/2018-05-06-datreant-downsize.md]: We do know ourselves how painful such an upgrade can be. The changes we plan simplify the code drastically and fully define the feature set of datreant. We don't plan to add major breaking changes in the future after this.

To ease transition you can also opt to continue using the older versions of the library for existing projects and use the new improved version for new projects using conda environments or pip virtual envs. (link to MDA blog post about both).

**Review by kain88-de — COMMENTED** [review]
some small comments. I might have time later this week to edit the post myself.

File: _posts/2018-05-06-datreant-downsize.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+---
+layout: post
+comments: true
+title: datreant downsizing
+---
+
+The first commit of what became `datreant` occurred over four years ago, and in the course of its development it's seen many iterations on what it is and how it works.
+The library didn't even start life as `datreant`: MDSynthesis was born first, with `datreant` becoming a more general library for handling datasets dispersed across a filesystem.
```
Comment (orbeckst) [inline _posts/2018-05-06-datreant-downsize.md]: link to MDSynthesis?

File: _posts/2018-05-06-datreant-downsize.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+## What will `datreant` look like now?
+
+The biggest change came in [Issue #145](https://github.com/datreant/datreant.core/pull/145).
+A directory is now marked as a Treant if it contains a `.datreant` directory.
+There are no UUIDs, which could be trouble when copying Treants around.
+Tags and Categories are stored individually in their own files in this directory, improving performance when either becomes large.
+This change also makes the `discover` machinery more easily able to find Treants,
+improving its performance greatly after this was identified as a bottleneck for larger Treant collections.
+
+`datreant` is also no longer a namespace package, and "limbs" have been elminated as a concept.
```
Comment (orbeckst) [inline _posts/2018-05-06-datreant-downsize.md]: eliminated

File: _posts/2018-05-06-datreant-downsize.md (hunk 2 above)
Comment (orbeckst) [inline _posts/2018-05-06-datreant-downsize.md]: Can you say something about how the new datreant will interact with MDSynthesis and datreant.data?

File: _posts/2018-05-06-datreant-downsize.md (hunk 1 above)
Comment (orbeckst) [reply _posts/2018-05-06-datreant-downsize.md]: Especially given the semantic meaning of "1.0.0" is a stable thing... 0.8.0 sends a mixed message.

File: _posts/2018-05-06-datreant-downsize.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+In an ideal world, we could roll out gradual changes to `datreant` in a controlled way, according to a clear project plan.
+None of our changes would be a huge disruption to users, and we'd do everything technically possible to ease the transition.
+But this project is small, with ~3 developers spending a few extra minutes on it every month.
+We don't have the bandwidth to do everything the way we want to.
+
+With that, we're releasing `datreant` 0.8.0 in the next few days, with a package on PyPI shortly after.
+Please uninstall all `datreant.*` packages from your environment before installing this, as the reversion from a namespace package will lead to unexpected behavior otherwise.
+
+If you are running existing projects using `datreant` and don't want to be affected by these changes
+we recommend that you pin your dependency to version 0.7.1.
```
Comment (orbeckst) [inline _posts/2018-05-06-datreant-downsize.md]: Is there any way to transition projects?

Can you outline 
- what needs to be done to move from old to new treants
- what changes in user code need to be made (e.g., don't use uuid)

File: blog/index.html [hunk 6]
```diff
@@ -0,0 +1,37 @@
+---
```
Comment (orbeckst) [inline blog/index.html:1]: I think you also need to add code to [/_includes/sidebar.html)](https://github.com/datreant/datreant.github.io/blob/master/_includes/sidebar.html) for the blog to show up. I just looked at the MDA pages and it has stuff like
```
      {% assign blogpagedir = site.paginate_path | split: ":" | first %}
      {% assign blogpagedir_len = blogpagedir | size %}
      {% assign pages_list = site.pages %}
      {% for node in pages_list %}
        {% if node.title != null %}
          {% if node.layout == "page" %}
            {% assign prefix = node.url | truncate: blogpagedir_len, '' %}
            {% if prefix != blogpagedir %}
              <a class="sidebar-nav-item{% if page.url == node.url %} active{% endif %}" href="{{ node.url }}">{{ node.title }}</a>
            {% endif %}
          {% endif %}
        {% endif %}
      {% endfor %}
```

**Review by orbeckst — COMMENTED** [review]
Minor comments. Good luck with 1.0.0!

File: _posts/2018-05-06-datreant-downsize.md (hunk 1 above)
Comment (dotsdl) [reply _posts/2018-05-06-datreant-downsize.md]: Agreed. We're releasing 1.0.

File: _posts/2018-05-06-datreant-downsize.md (hunk 2 above)
Comment (dotsdl) [reply _posts/2018-05-06-datreant-downsize.md]: Done!

File: _posts/2018-05-06-datreant-downsize.md (hunk 5 above)
Comment (dotsdl) [reply _posts/2018-05-06-datreant-downsize.md]: Done.

File: _posts/2018-05-06-datreant-downsize.md (hunk 2 above)
Comment (dotsdl) [reply _posts/2018-05-06-datreant-downsize.md]: I hope I conveyed things well. Let me know what you think.

File: _posts/2018-05-06-datreant-downsize.md (hunk 3 above)
Comment (dotsdl) [reply _posts/2018-05-06-datreant-downsize.md]: Done!

File: _posts/2018-05-06-datreant-downsize.md (hunk 4 above)
Comment (dotsdl) [reply _posts/2018-05-06-datreant-downsize.md]: Thanks!

File: blog/index.html (hunk 6 above)
Comment (dotsdl) [reply blog/index.html:1]: It shows up...I believe I added it? Don't recall...

**Comment by dotsdl** [comment]
Ready to merge if y'all are. Then we can start this show. XD

**Comment by richardjgowers** [comment]
@dotsdl looks good to me!

**Review by kain88-de — APPROVED** [review]

**Comment by orbeckst** [comment]
http://datreant.org/blog/ does not have CSS styling (tested in FF and Chrome)

**Comment by dotsdl** [comment]
@orbeckst I noticed that; I tried looking through what we did for MDAnalysis's blog but can't tell what's missing. Will have to check it out tonight to fix unless someone can figure out the missing piece.

**Comment by orbeckst** [comment]
I think you have to change the paths to your CSS to absolute (or use some other tricks) – at the moment you have
```html
 <link rel="stylesheet" href="public/css/poole.css">
  <link rel="stylesheet" href="public/css/syntax.css">
  <link rel="stylesheet" href="public/css/hyde.css">
```
and as soon as you're under `.../blog/`, these paths are not found anymore.

**Comment by orbeckst** [comment]
```
_includes/head.html:  <link rel="stylesheet" href="{{ site.baseurl }}/public/css/poole.css">
```

**Pull request merged**